### PR TITLE
Burger menu redesign

### PR DESCRIPTION
**Pull request by bbecquet**
## Description
Redesign the burger menu.
 - remove most of the menu content, which is considered confusing or outdated
 - new squared design of the button on map and new icons (burger, close circle and items)
 - new partly transparent overlay, instead of a nearly opaque one

For now, the mobile version won't match the final design. It would require a complex drawer component that we don't want to clone in Erdapfel before we can use the real Design System components. The compromise is to use a sliding panel, like on desktop and like the current prod version on mobile.

## Why
Modernization + simplification + matching the next Search design

## Screenshots
|Before|After|
|---|---|
|www qwant com_maps_|localhost_3000_favs_ (1)|

Peek 26-01-2021 15-48

### CODE REVIEW

File: src/panel/Menu.jsx [hunk 1]
```diff
@@ -36,58 +37,47 @@ const Menu = () => {
...
-              <i className="menu__panel__action__icon icon-zap" />
-              <span>{_('How to contribute', 'menu')}</span>
-            </a>
-          </div>
-
-          {menuItems.map(menuItem => <MenuItem key={menuItem.sectionName} menuItem={menuItem} />)}
+            }}
+            icon={<Heart width={16} color={PINK_DARK} />}
+          >
+            {_('Favorites', 'menu')}
```
Comment (amatissart) [inline src/panel/Menu.jsx:65]: The mockup specified "My favorites", which is consistent with the title used on the FavoritesPanel header.

Comment (bbecquet) [reply src/panel/Menu.jsx]: Fixed.

File: src/components/ui/icons.js [hunk 2]
```diff
@@ -1,2 +1,8 @@
 export { default as Heart } from '../../../public/images/feather/heart.svg';
 export { default as Magnifier } from '../../../public/images/magnifier-dark.svg';
+
+export { default as IconLightbulb } from '../../../public/images/remix/lightbulb-line.svg';
```
Comment (G-Ray) [inline src/components/ui/icons.js:4]: :+1:

File: src/scss/includes/menu.scss [hunk 3]
```diff
@@ -160,20 +73,32 @@
...
-
-    &__line {
-      margin: 2px auto;
+    right: $spacing-s;
+    top: $spacing-xl;
+    width: 40px;
+    height: 40px;
+    border-radius: 8px;
+    box-shadow: $shadow;
+    color: $grey-semi-darkness;
```
Comment (G-Ray) [inline src/scss/includes/menu.scss:82]: `grey-semi-darkness` should be set on mobile too :lipstick:

File: src/panel/Menu.jsx [hunk 4]
```diff
@@ -36,58 +37,47 @@ const Menu = () => {
...
-            >
-              <img
-                className="menu__panel__action__icon"
-                src="./statics/images/magnifier-dark.svg" alt=""
-              />
-              <span>{_('Search', 'menu')}</span>
-            </button>
-            {isDirectionActive &&
-              <button className="menu__panel__action" onClick={() => {
-                Telemetry.add(Telemetry.MENU_ITINERARY);
```
Comment (G-Ray) [inline src/panel/Menu.jsx:66]: if we get rid of this specific event, we can remove the `menu_itinerary` event from the `telemetry` module

File: src/panel/Menu.jsx [hunk 5]
```diff
@@ -21,9 +20,11 @@ const Menu = () => {
     };
   }, []);
 
-  const open = () => {
-    Telemetry.add(Telemetry.MENU_CLICK);
-    setIsOpen(true);
+  const toggle = () => {
+    if (!isOpen) {
+      Telemetry.add(Telemetry.MENU_CLICK);
```
Comment (G-Ray) [inline src/panel/Menu.jsx:25]: Ok, so we can expect 2 times less registered events from now IIUC

File: src/panel/menu/MenuItem.jsx [hunk 6]
```diff
@@ -1,27 +1,23 @@
...
-    {links.map(link => <a
-      key={link.uri}
-      className="menu__panel__link"
-      href={link.uri} rel="noopener noreferrer" target="_blank"
-    >
-      {_(link.name)}
-    </a>)}
-  </div>}
-</div>;
+const MenuItem = ({ icon, children, href, onClick, outsideLink }) =>
```
Comment (G-Ray) [inline src/panel/menu/MenuItem.jsx:6]: :+1:

File: src/panel/Menu.jsx (hunk 5 above)
Comment (bbecquet) [reply src/panel/Menu.jsx:25]: Yep, and nobody cares :)

**Review by G-Ray — APPROVED** [review]

**Pull request merged**